wweval/R/ms_fig2.R

```diff
@@ -217,6 +230,154 @@ make_fig2_ct <- function(ww_quantiles,
+#' @param ms_fig_dir A string indicating where to save the figure
+#' @param n_forecast_days An integer indicating the number of days to show the
+#' forecast for, default is `28`
+#' @param n_calib_days An integer indicating the number of days to show the
+#' calibration data for, default is `90`
+#' @param max_n_site_labs_to_show An integer indicating the maximum number
+#' of site-labs to show in the figure, default is `3`
+#' @param site_lab_names_to_show a vector of character strings indicating
+#' the site lab names to be displayed in the plot. If NULL, the first
+#' `max_n_site_labs_to_show` or all are displayed. Default is `NULL`.
+#'
+#' @return A ggplot object containing a faceted horizontal plot of the
+#' calibrated and forecasted wastewater concentrations for 3 or fewer
+#' site-lab combinations for a single state
+#' @export
+make_fig2_ct_supp <- function(ww_quantiles,
+                              loc_to_plot,
+                              date_to_plot,
+                              ms_fig_dir,
+                              n_forecast_days = 28,
+                              n_calib_days = 90,
+                              max_n_site_labs_to_show = 3,
+                              site_lab_names_to_show = NULL) {
+  if (!is.null(site_lab_names_to_show)) {
+    ww_quantiles <- ww_quantiles |>
+      dplyr::filter(site_lab_name %in% c(site_lab_names_to_show))
+  } else {
+    ww_quantiles <- ww_quantiles |>
+      dplyr::filter(lab_site_index <= !!max_n_site_labs_to_show)
+  }
+
+  ww <- ww_quantiles |>
+    dplyr::filter(location == !!loc_to_plot) |>
+    dplyr::filter(forecast_date == !!date_to_plot) |>
+    dplyr::filter(
+      date <= forecast_date + lubridate::days(!!n_forecast_days),
+      date >= forecast_date - lubridate::days(!!n_calib_days)
+    )
+
+
+
+  stopifnot(
+    "This function is meant for one location" =
+      length(unique(ww$location)) <= 1
+  )
+
+  quantiles_wide <- ww |>
+    dplyr::mutate(log_conc = log(value)) |>
+    dplyr::filter(quantile %in% c(0.025, 0.25, 0.5, 0.75, 0.975)) |>
+    tidyr::pivot_wider(
+      id_cols = c(
+        location, site_lab_name, forecast_date, period, scenario,
+        date, eval_data, calib_data, below_LOD, flag_as_ww_outlier
+      ),
+      names_from = quantile,
+      values_from = log_conc
+    ) |>
+    dplyr::mutate(
+      model = "ww",
+      observation_status =
+        dplyr::case_when(
+          flag_as_ww_outlier == 1 ~ "outlier",
+          below_LOD == 1 ~ "below LOD",
+          TRUE ~ "standard"
+        )
+    )
+
+
+
+  colors <- plot_components()
+  # Set ribbon and line color for model fit, in this case is always ww model
+  model_color <- as.character(colors$model_colors["ww"])
+
+  p <- ggplot(quantiles_wide) +
+    geom_point(aes(x = date, y = log(eval_data)),
+      fill = "white", size = 1, shape = 21,
+      show.legend = FALSE
+    ) +
+    geom_point(
+      aes(
+        x = date, y = log(calib_data),
```

Comment:
better to use `scale_y_continuous(transform="log10")` as it will result in better axis labels. Can exponentiate the log scale ribbon bounds as needed.

Reply:
Not essential for this PR.

Reply:
So the reason for doing the transform here is that the model output is on the log scale, the data in natural scale (I know this is sort of confusing -- the wwinference package by default returns everything in ww concentrations in log scale and I didn't change it back

Reply:
Created #219